### DISCUSSION TITLE
Why isn't my verifiy script working?

### DISCUSSION THREAD

**Discussion by pedrohba1** [foundry-rs/foundry#5130] [Help]
I've written this script to verify an contract that receives an address, bytes32 and another address in the constructor:

```
	forge verify-contract \
		--chain-id 80001 \
		--watch \
	--constructor-args "$(cast abi-encode 'constructor(address,bytes32,address)' '0x40193c8518BB267228Fc409a613bDbD8eC5a97b3' "$(cast --format-bytes32-string 'ca98366cc7314957b8c012c72f05aeeb')" '0x326C977E6efc84E512bB9C30f76E30c160eD06FB')" \
		--etherscan-api-key ${MUMBAI_SCAN_KEY} \
		--compiler-version v0.8.13+commit.abaa5c0e \
		0x323f8121af4e4b064b658666c1ad1e8c5c2d654e \
		src/ModeratioWithConsumer.sol:ModeratioWithConsumer
```

However, mumbai scan is always failing to verify. I'm pretty sure all the values are right. Am I missing something on enconding these values?

**Comment by canu0205** [comment]
Oh, I got the same issue at Sepolia testnet... Did you solve it?

**Reply by pedrohba1** [reply]
No, not yet. Still looking for a solution. Maybe using the json alternative. But I don't know how to yet.

**Comment by pyromaus** [comment]
Hey y'all, for the compiler version parameter try putting it in your `toml` instead of passing it to the terminal:
```toml
solc = "0.8.19"
```
or whichever version.

The terminal won't notice changes in your .env until you do `source .env` (in case anyone forgot) run that after adding the API key.
Try deleting the out folder and try again.
Go to terminal, and do `foundryup` then try again.
